src/bun.js/webcore/response.zig

```diff
@@ -974,9 +1139,24 @@ pub const Fetch = struct {
         }
 
         pub fn callback(task: *FetchTasklet, result: HTTPClient.HTTPClientResult) void {
-            task.response_buffer = result.body.?.*;
+            task.mutex.lock();
+            defer task.mutex.unlock();
             task.result = result;
-            task.javascript_vm.eventLoop().enqueueTaskConcurrent(task.concurrent_task.from(task, .manual_deinit));
+            task.body_size = result.body_size;
+
+            const success = result.isSuccess();
+            task.response_buffer = result.body.?.*;
+
+            if (success) {
+                _ = task.scheduled_response_buffer.write(task.response_buffer.list.items) catch @panic("OOM");
+            }
+
+            if (!task.has_schedule_callback.load(.Acquire)) {
```

Comment:
This is a race condition. The value can change between this load and the store below it. Instead, you want to use one of the compare and exchange ones.

Reply:
https://github.com/oven-sh/bun/pull/4127/commits/76b4e7bfdb810d77e2e8e36213563b00d77edb6f changed for compare and swap